FunctionalTableData/TableSectionChangeSet.swift

```diff
@@ -134,7 +134,11 @@ public final class TableSectionChangeSet {
 				if oldSectionIndex < old.count && new[newSectionIndex].key == old[oldSectionIndex].key {
 					// Skip over equal sections
 					repeat {
-						compareRows(newRows: &newRows, oldRows: &oldRows, oldSectionIndex: oldSectionIndex, newSectionIndex: newSectionIndex)
+						if headerOrFooterChanged(oldSectionIndex: oldSectionIndex, newSectionIndex: newSectionIndex) {
+							reloadedSections.insert(newSectionIndex)
+						} else {
+							compareRows(newRows: &newRows, oldRows: &oldRows, oldSectionIndex: oldSectionIndex, newSectionIndex: newSectionIndex)
```

Comment:
I believe this should be

```swift
compareSectionsHeadersFooters()
compareRows()
```

otherwise if they are equal the rows are entirely skipped right?

Reply:
`headerOrFooterChanged(...)` returns true if the state of the header or the state of the footer has changed. In either of those cases, we need to reload the whole section to update the appropriate view, so we can avoid comparing rows. 
We only need to compare rows if `headerOrFooterChanged(...)` returns false.